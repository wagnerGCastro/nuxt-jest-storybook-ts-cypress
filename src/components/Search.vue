<template>
  <form data-testid="search-form" class="search-form d-flex" @submit.prevent="doSearch">
    <input v-model="term" class="w-full border rounded-md" type="search" placeholder="Search" />
    <v-icon dense class="icon search-form__icon">mdi-magnify</v-icon>
  </form>
</template>

<script>
export default {
  name: 'Search',
  data() {
    return {
      term: '',
    };
  },
  watch: {
    term() {
      if (this.term === '') {
        this.doSearch();
      }
    },
  },
  methods: {
    doSearch() {
      this.$emit('doSearch', { term: this.term });
    },
  },
};
</script>

<style scoped>
.search-form {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  position: relative;
  border: 1px solid #ccc;
  border-radius: 5px;
  padding: 2px 5px;
}

.search-form [type='search'] {
  color: #999;
}

.search-form__icon {
  position: absolute;
  right: 5px;
  outline: none;
}
</style>
