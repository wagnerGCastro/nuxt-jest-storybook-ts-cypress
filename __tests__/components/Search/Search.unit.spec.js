import Vuetify from 'vuetify';
import { mount } from '@vue/test-utils';
import Search from '@/components/Search';

describe('Search - unit', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('should mount the component', () => {
    const wrapper = mount(Search, { vuetify });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.vm).toBeDefined();
  });

  it('should emit search event when form is submit', async () => {
    const wrapper = mount(Search, { vuetify });
    const term = 'term to search';

    await wrapper.find('input[type="search"]').setValue(term);
    await wrapper.find('form').trigger('submit');

    // expect(wrapper.emitted().doSearch).toBeFalsy();
    expect(wrapper.emitted().doSearch).toBeTruthy();
    expect(wrapper.emitted().doSearch.length).toBe(1);
    expect(wrapper.emitted().doSearch[0]).toEqual([{ term }]);
  });

  it('should emit search event when search input is cleared', async () => {
    const wrapper = mount(Search, { vuetify });
    const term = 'termo para busca';
    const input = wrapper.find('input[type="search"]');

    await input.setValue(term);
    await input.setValue('');

    expect(wrapper.emitted().doSearch).toBeTruthy();
    expect(wrapper.emitted().doSearch.length).toBe(1);
    expect(wrapper.emitted().doSearch[0]).toEqual([{ term: '' }]);
  });
});
