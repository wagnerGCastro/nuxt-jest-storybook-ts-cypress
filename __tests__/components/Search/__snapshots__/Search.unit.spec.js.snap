// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search - unit should mount the component 1`] = `
<form
  class="search-form d-flex"
  data-testid="search-form"
>
  <input
    class="w-full border rounded-md"
    placeholder="Search"
    type="search"
  />
   
  <i
    aria-hidden="true"
    class="v-icon notranslate icon search-form__icon v-icon--dense mdi mdi-magnify theme--light"
  />
</form>
`;
